### PR TITLE
Fix missing override warnings in ServerConnectionAcceptorTest.cpp

### PR DESCRIPTION
**Pull request by alexmalyshev**
MOCK_METHOD*() was being used to mock virtual methods.  There's nothing wrong
with this, it's just gcc/clang will warn about missing an override qualifier.

### CODE REVIEW

File: test/ServerConnectionAcceptorTest.cpp [hunk 1]
```diff
@@ -23,29 +23,45 @@ class MockConnectionHandler : public ConnectionHandler {
   void setupNewSocket(
       std::shared_ptr<FrameTransport> frameTransport,
       ConnectionSetupPayload setupPayload) override {
-    doSetupNewSocket(std::move(frameTransport), setupPayload);
+    setupNewSocket_(std::move(frameTransport), setupPayload);
+  }
+
+  bool resumeSocket(
```
Comment (lehecka) [inline test/ServerConnectionAcceptorTest.cpp]: override keyword?

File: test/ServerConnectionAcceptorTest.cpp [hunk 2]
```diff
@@ -23,29 +23,45 @@ class MockConnectionHandler : public ConnectionHandler {
...
+      std::shared_ptr<FrameTransport> transport,
+      ResumeParameters params) override {
+    return resumeSocket_(std::move(transport), params);
+  }
+
+  void connectionError(
```
Comment (lehecka) [inline test/ServerConnectionAcceptorTest.cpp]: ditto

**Comment by lexs** [comment]
This is unfortunate because gtest hasn't been updated in a long time and doesn't support the override qualifier (or move-only types etc). MOCK_METHOD is really for mocking virtual methods.

Instead of adding more boilerplate could we either disable the warning in `test/` or add a macro that does the boilerplate for you?

File: test/ServerConnectionAcceptorTest.cpp (hunk 1 above)
Comment (alexmalyshev) [inline test/ServerConnectionAcceptorTest.cpp]: It's on the `resumeSocket()` method already.

File: test/ServerConnectionAcceptorTest.cpp [hunk 3]
```diff
@@ -23,29 +23,45 @@ class MockConnectionHandler : public ConnectionHandler {
   void setupNewSocket(
       std::shared_ptr<FrameTransport> frameTransport,
       ConnectionSetupPayload setupPayload) override {
-    doSetupNewSocket(std::move(frameTransport), setupPayload);
+    setupNewSocket_(std::move(frameTransport), setupPayload);
+  }
+
+  bool resumeSocket(
+      std::shared_ptr<FrameTransport> transport,
+      ResumeParameters params) override {
```
Comment (alexmalyshev) [inline test/ServerConnectionAcceptorTest.cpp]: Right here

**Comment by alexmalyshev** [comment]
> Instead of adding more boilerplate...

I'll look into disabling the warning for `test/`.  I don't feel like adding a macro that'll need to be #included by every test file that uses mocks...

**Comment by lexs** [comment]
Thank you!

**Pull request merged**